hera_sim/noise.py

```diff
@@ -7,24 +7,26 @@
 import aipy
 import os
 from .data import DATA_PATH
-HERA_TSKY_VS_LST_NPZ = os.path.join(DATA_PATH, 'HERA_Tsky_vs_LST.npz')
-
-npz = np.load(HERA_TSKY_VS_LST_NPZ) # Tsky vs fq/lst from Beardsley, beam v XXX, GSM v XXX
-fqs = npz['freqs'] / 1e3
-lsts = npz['lsts'] / 12. * np.pi
-lsts = np.concatenate([lsts[-10:]-2*np.pi, lsts, lsts[:10]+2*np.pi])
-HERA_Tsky_xx = npz['HERA_Tsky'][0].T
-HERA_Tsky_yy = npz['HERA_Tsky'][1].T
-HERA_Tsky_xx = np.concatenate([HERA_Tsky_xx[-10:], HERA_Tsky_xx, HERA_Tsky_xx[:10]])
-HERA_Tsky_yy = np.concatenate([HERA_Tsky_yy[-10:], HERA_Tsky_yy, HERA_Tsky_yy[:10]])
+from .interpolators import Tsky, _check_path
+from .defaults import _defaults
+
+HERA_TSKY_VS_LST_NPZ = os.path.join(DATA_PATH, 'HERA_Tsky_Reformatted.npz')
+
 HERA_Tsky_mdl = {}
-HERA_Tsky_mdl['xx'] = RectBivariateSpline(lsts, fqs, HERA_Tsky_xx, kx=4, ky=4)
-HERA_Tsky_mdl['yy'] = RectBivariateSpline(lsts, fqs, HERA_Tsky_yy, kx=4, ky=4)
+HERA_Tsky_mdl['xx'] = Tsky(HERA_TSKY_VS_LST_NPZ, pol='xx')
+HERA_Tsky_mdl['yy'] = Tsky(HERA_TSKY_VS_LST_NPZ, pol='yy')
 
+@_defaults
+def _get_hera_bm_poly(datafile='HERA_H1C_BEAM_POLY.npy'):
+    """
+    Method for getting HERA bandpass polynomial coefficients. This should be
+    replaced in the future.
+    """
+    datafile = _check_path(datafile)
+    return np.load(datafile)
 
-HERA_BEAM_POLY = np.array([8.07774113e+08, -1.02194430e+09,
-                           5.59397878e+08, -1.72970713e+08, 3.30317669e+07, -3.98798031e+06,
-                           2.97189690e+05, -1.24980700e+04, 2.27220000e+02])  # See HERA Memo #27
+# XXX I don't like this. Also figure out what to do about omega_p.
+HERA_BEAM_POLY = _get_hera_bm_poly()
```

Comment:
It turns out this doesn't work the way I'd like it to--`HERA_BEAM_POLY` just gets defined at runtime and then is insensitive to changing defaults.

Reply:
Isn't that what your decorator is supposed to take care of?

Reply:
It is, but it looks like the decorator doesn't work for situations like this. However, I did figure out a workaround. So currently an object like `HERA_BEAM_POLY` gets permanently set to whatever the initial default setting is (currently H1C). I think that's okay, though, because we shouldn't be storing data in the code anyway; I see their existence from here onward as more of a legacy feature, kept primarily for historical purposes.